cmd/yggd/client.go

```diff
@@ -11,41 +11,123 @@ import (
 	"git.sr.ht/~spc/go-log"
 	"github.com/google/uuid"
 	"github.com/redhatinsights/yggdrasil"
+	"github.com/redhatinsights/yggdrasil/internal/config"
 	"github.com/redhatinsights/yggdrasil/internal/transport"
+	"github.com/redhatinsights/yggdrasil/internal/work"
 )
 
 type Client struct {
 	t transport.Transporter
-	d *dispatcher
+	d *work.Dispatcher
 }
 
+// NewClient creates a new Client configured with dispatcher and transporter.
+func NewClient(dispatcher *work.Dispatcher, transporter transport.Transporter) *Client {
+	return &Client{
+		d: dispatcher,
+		t: transporter,
+	}
+}
+
+// Connect starts a goroutine receiving values from the client's dispatcher and
+// transmits the data using the transporter.
 func (c *Client) Connect() error {
+	if c.t == nil {
+		return fmt.Errorf("cannot connect client: missing transport")
+	}
+
+	// Connect the Dispatcher
+	if err := c.d.Connect(); err != nil {
+		return fmt.Errorf("cannot connect dispatcher: %w", err)
+	}
+
+	// start receiving values from the dispatcher and transmit them using the
+	// provided transporter.
+	go func() {
+		for msg := range c.d.Outbound {
+			code, metadata, data, err := c.SendDataMessage(&msg.Data, msg.Data.Metadata)
+			if err != nil {
+				log.Errorf("cannot send data message: %v", err)
+				continue
+			}
+			msg.Resp <- struct {
```

Comment:
Shouldn't this struct have a type, it's a bit weird

Reply:
Good point. I refactored this out into a separate data type.